react/utils/pages.ts

```diff
@@ -125,7 +126,16 @@ export function navigate(history: History | null, pages: Pages, options: Navigat
   return false
 }
 
-export function scrollTo(options: ScrollToOptions) {
+export function scrollTo(options: RenderScrollToOptions) {
+  const { elementId } = options
+  if (elementId) {
+    const scrollAnchor = document.querySelector(`#${elementId}`)
+    if (scrollAnchor) {
```

Comment:
I do not love the fact that `elementId`, when set, ignores the other options. As I've commented in person, I would use [`element.getBoundingClientRect`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect), `window.scrollX` and `window.scrollY` to still allow offsets.

Alternatively, I would make `RenderScrollToOptions` not be an extension of `ScrollToOptions`, but an union, which makes it clear that either one or the other configuration may be applied:

```
type AchoredScrollOptions = { elementId: string }
type RenderScrollToOptions = ScrollToOptions | AnchoredScrollOptions
``` 

Still taking into account my name change suggestion in the comment above.

Reply:
Made the scroll relative to the element with the ID